addons/l10n_jo_edi/models/account_edi_xml_ubl_21_jo.py

```diff
@@ -0,0 +1,360 @@
+            'company_id': partner.vat,
+            'tax_scheme_vals': {'id': 'VAT'},
+        }]
+
+    def _get_partner_party_legal_entity_vals_list(self, partner):
+        return [{
+            'registration_name': partner.name,
+        }]
+
+    def _get_partner_contact_vals(self, partner):
+        return {}
+
+    def _get_empty_party_vals(self):
+        return {
+            'postal_address_vals': {'country_vals': {'identification_code': 'JO'}},
+            'party_tax_scheme_vals': [{'tax_scheme_vals': {'id': 'VAT'}}],
+        }
+
+    def _get_partner_party_vals(self, partner, role):
+        vals = super()._get_partner_party_vals(partner, role)
+        vals['party_name_vals'] = []
+        if role == 'supplier':
+            vals['party_identification_vals'] = []
+        return vals
+
+    def _get_delivery_vals_list(self, invoice):
+        return []
+
+    def _get_invoice_payment_means_vals_list(self, invoice):
+        if invoice.move_type == 'out_refund':
+            return [{
+                'payment_means_code': 10,
+                'payment_means_code_attrs': {'listID': "UN/ECE 4461"},
+                'instruction_note': invoice.ref.replace('/', '_'),
+            }]
+        else:
+            return []
+
+    def _get_invoice_payment_terms_vals_list(self, invoice):
+        return []
+
+    def _get_invoice_tax_totals_vals_list_helper(self, invoice, taxes_vals):
+        tax_totals_vals = {
+            'currency': JO_CURRENCY,
+            'currency_dp': self._get_currency_decimal_places(),
+            'tax_amount': 0,
+            'tax_subtotal_vals': [],
+        }
+        for grouping_key, vals in taxes_vals['tax_details'].items():
+            if grouping_key['tax_amount_type'] != 'fixed':
+                subtotal = {
+                    'currency': JO_CURRENCY,
+                    'currency_dp': self._get_currency_decimal_places(),
+                    'taxable_amount': vals['raw_base_amount'],
+                    'tax_amount': vals['raw_tax_amount'],
+                    'tax_category_vals': vals['_tax_category_vals_'],
+                }
+                tax_totals_vals['tax_subtotal_vals'].append(subtotal)
+                tax_totals_vals['tax_amount'] += subtotal['tax_amount']
+
+        return [tax_totals_vals]
+
+    def _get_invoice_tax_totals_vals_list(self, invoice, taxes_vals):
+        if invoice._get_jo_invoice_type_number() in [1, 2]:
+            return []
+
+        vals = self._get_invoice_tax_totals_vals_list_helper(invoice, taxes_vals)
+        if invoice._get_jo_invoice_type_number() != 4:
+            vals[0]['tax_subtotal_vals'] = []
+        return vals
+
+    def _get_invoice_line_item_vals(self, line, taxes_vals):
+        product = line.product_id
+        description = line.name and line.name.replace('\n', ', ')
+        return {
+            'name': product.name or description,
+        }
+
+    def _get_document_allowance_charge_vals_list(self, invoice, taxes_vals):
```

Comment:
I see this method is normally used for the global discounts (e.g. a negative invoice line in Odoo), not the discounts per line.  (weird that it does not give errors then)

Reply:
Please provide a docstring here: 
""" For JO UBL the document allowance charge vals needs to be the sum of the line discounts. """